### DISCUSSION TITLE
run on localhost

### DISCUSSION THREAD

**Discussion by dvmarinoff** [dvmarinoff/Auuki#155] [Q&A]
@mesterumanele, I moved you post to discussions:

> Hello,
> i love your app, it is just amazing! 
> My question/note is this: having all these "The web version is free to use and open source. It is developed as a static PWA, has
> zero dependancies and is completly independant to run." but still works with internet beats the purpose of it. if i have internet i
> can run intelligent-cycling or mywhoosh, no? I thought that your app can be run from localhost instead of internet! Then all of the > above are true. This is just a thought, i'm not sure what's behind the scenes, i'm ignorant about programing, so if my note
> doesn't make sense, just ignore it! Thank you

**Comment by dvmarinoff** [comment]
@mesterumanele Hi and welcome to the project!

I am not quite sure what is your question, but I'd assume it's about running the app locally or offline.
So Yes, you can run the app on localhost. You just need to setup a local https server or use the one configured in the build and development configuration of the app. If you are a programmer that's easy just type 2 or 3 lines in the console.

```
$ git clone https://github.com/dvmarinoff/Flux.git
$ npm install
$ npm start
```

But that requires that you have both `git` and `npm` installed on your machine, but that's the case practically every programmer out there.

There is also the option to install the app and run it offline, but that requires someone to maintain the 'offline use' configuration, and currently there isn't much interest in that feature so the configuration is a bit outdated and does not work.

**Comment by dvmarinoff** [comment]
Just to further clarify, the `has zero dependancies and is completly indipendant to run` refers to the app requires no external libraries or programming frameworks to run. Meaning the app uses simple technologies in their pure form, that is: html, css and javascript. So the barrier to entry for other programmers (or researchers) that wish to play with the code or add their own features is very low.
And moreover the project is self contained and does not depend on the faith of some other project.

**Reply by mesterumanele** [reply]
Thank you for the reply. I see usefulness of your project to run it offline (when no internet, at cottage no wifi, things like that). Your answer makes sense, if it's offline it's a barrier to contribute online/etc... i would like to further explore the part from your first answer were you said, yes, is possible:
- i install git and npm (again, sorry, i'm not a programmer, but i'll do my best, even willing to learn it)
-  "use the one configured in the build and development configuration of the app" means i download the <code-zip> 
- "that's easy just type 2 or 3 lines in the console." means comand promp/cmd, if i'm correct
- this seems to be do'able and then would be no need to have "someone to maintain the 'offline use' " 
Thank you again,

**Reply by dvmarinoff** [reply]
From here you can download git installer for windows: https://git-scm.com/download/win
And from here node/npm : https://nodejs.org/en/download

You can checkout these instruction here: https://olanetsoft.medium.com/setting-up-your-development-environment-git-and-node-948613b1239a

After you have those 2 installed you'd need to use the git-console to run this line
```$ git clone https://github.com/dvmarinoff/Flux.git```

Than enter the directory with `$ cd Flux`
Than run the other 2 commands:
```
$ npm install
$ npm start
```

This will allow you to run the app on a desktop/laptop machine.

**Comment by mesterumanele** [comment]
Appreciate your details
Thanks

**Comment by mesterumanele** [comment]
Hello,
all instructions were spot on: i made it work from http://localhost:1234

I still have a question, please: if for example I want to try the ios
webBLE app, should I enter on webBLE the address: http://192.168.x.x:1234?
where 192.168.x.x is the ip of where 'localhost:1234' is running. Or
similarly, from Mac intel chrome browser i try http://192.168.x.x:1234 but
nothing happens: no error, no connection

Thank you,
m

**Reply by dvmarinoff** [reply]
I am not sure I follow, but iOS is not supported. The browsers on iOS don't have web bluetooth or any other app-like feature, so web apps are really limited on iOS.

To open the app from another device you need to find the external IP address of the device that's running the server, and open that instead. I don't know how to do this on Windows, but maybe a google search will help.

**Comment by mesterumanele** [comment]
Hi, thank you for the reply!

I understand from your posting that ios doesn't have support; but in the
same posting you suggest to try webBLE app, it might work when connecting
to flux-web.vercel.app. you suggest webBLE might work + other browser
(chrome/yes, edge/yes, opera/yes chromeAndroid/yes, samsungInternet/yes).

Basically, i'm trying to connected with your suggested(webBLE)+approved
browser to my local host instead of the web app flux-web.vercel.app

now, if local flux works on localhost:1234 then i should be able to connect
from another browser inside the lan, with the address of that localhost, as
you suggested... for example localhost is 192.168.2.36... my local flux
works on the computer where flux is running, on 192.168.2.36:1234 but when
i try from a different computer, with edge for example, to try to connect
to 192.168.2.36:1234 nothing happens: no error, no flux dashboard, just
white empty page... this is my question/issue... again, it's not major
issue, just a note. I wonder if localhost flux is made just for localhost?
or should an outside browser use a different port than 1234? i did try 80
but the same result, white page with no error on it...

Thank you for your time and apologize if I'm confusing you or because I'm
confused.
Thanks
